src/schemas/Entity.js

```diff
@@ -48,12 +73,18 @@ export default class EntitySchema {
     return this._mergeStrategy(entityA, entityB);
   }
 
-  normalize(input, parent, key, visit, addEntity) {
+  normalize(input, parent, key, visit, addEntity, visitedEntities) {
+    const stringifiedInput = this.stringifySafe(input);
+    if (visitedEntities[stringifiedInput]) {
```

Comment:
Why do you have to stringify instead of doing an exact equality check?

Something like `Object.values(visitedEntities).some(entity => entity === input)`